backend/internal/idp/file_based_store.go

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package idp
+
+import (
+	"errors"
+
+	"github.com/asgardeo/thunder/internal/system/file_based_runtime/entity"
+	"github.com/asgardeo/thunder/internal/system/log"
+)
+
+type idpFileBasedStore struct {
+	storage entity.StoreInterface
+}
+
+// CreateIdentityProvider implements idpStoreInterface.
+func (f *idpFileBasedStore) CreateIdentityProvider(idp IDPDTO) error {
+	idpKey := entity.NewCompositeKey(idp.ID, entity.KeyTypeIDP)
+	return f.storage.Set(idpKey, &idp)
+}
+
+// DeleteIdentityProvider implements idpStoreInterface.
+func (f *idpFileBasedStore) DeleteIdentityProvider(id string) error {
+	return errors.New("DeleteIdentityProvider is not supported in file-based store")
```

Comment:
can we return &filebasedruntime.ErrorImmutableResourceDeleteOperation

Same as we return from service layer